### ISSUE TITLE
Explicit graph kernels

### ISSUE THREAD

**Issue by congrendai** [ysig/GraKeL#91]
Hi, as far as I know, the explicit graph kernels can make a feature vector for each graph so that I can use the feature vectors to do any kinds of ML algorithms. Therefore, I am confused about how to create new kernels that don't do the inner product between every pairs of feature vectors and just keep the feature vectors for graphs. Could you kindly provide an example for such as shortest-path kernel that can return feature vectors matrix without calculating inner product, please?

**Comment by giannisnik** [comment]
Hi @congrendai ,

For some kernels where the mapping to the feature space is explicit, you can indeed obtain such features. For instance, for the Weisfeiler-Lehman subtree kernel, you can obtain the features for the different iterations as follows:
```
from grakel.datasets import fetch_dataset
from grakel.kernels import WeisfeilerLehman, VertexHistogram

dataset = fetch_dataset("MUTAG", verbose=False)
Gs = dataset.data
wl_kernel = WeisfeilerLehman(n_iter=3,  base_graph_kernel=VertexHistogram)
K = wl_kernel.fit_transform(Gs)
features_it_0 = wl_kernel.X[0].X
features_it_1 = wl_kernel.X[1].X
features_it_2 = wl_kernel.X[2].X
features_it_3 = wl_kernel.X[3].X
```
In this example, `features_it_0`, `features_it_1`, `features_it_2`, `features_it_3` are all sparse matrices where the number of rows is equal to the number of graphs (i.e., `len(Gs)`) and the number of columns is equal to the number of different labels assigned to the nodes of all graphs at that iteration.

For the shortest path kernel, you can obtain the features as follows:
```
from grakel.datasets import fetch_dataset
from grakel.kernels import ShortestPath

dataset = fetch_dataset("MUTAG", verbose=False)
Gs = dataset.data
sp_kernel = ShortestPath(with_labels=True)
K = sp_kernel.fit_transform(Gs)
features_sp = sp_kernel._phi_X
```
where `features_sp` is a matrix that contains the features produced by the shortest path kernel.
